packages/cursorless-vscode/src/keyboard/buildSuffixTrie.ts

```diff
@@ -0,0 +1,115 @@
+import { isEqual, range, sortBy, uniqWith } from "lodash";
+import TrieSearch, { TrieSearchOptions } from "trie-search";
+
+export interface KeyValuePair<T> {
+  key: string;
+  value: T;
+}
+
+interface BuildTrieReturn<T> {
+  trie: TrieSearch<KeyValuePair<T>>;
+  conflicts: KeyValuePair<T>[][];
+}
+type InternalEntryType<T> = {
+  isTopLevel: boolean;
+  key: string;
+  value: T;
+  id: number;
+};
+
+/**
+ * Build a trie containing all the keymaps, as well as non-conflicting suffixes
+ * of each key. Also returns a list of all the conflicting keymaps. If any key is
+ * a prefix of any other key, we consider them to be conflicting.
+ * @param keyMaps The keymaps to build the trie from
+ * @returns The trie, and a list of conflicting keymaps
+ */
+export function buildSuffixTrie<T>(entries: [string, T][]): BuildTrieReturn<T> {
```

Comment:
this is so much clearer. thanks for taking the time to do it and document it nicely. and test it so well. I have WAY more confidence now that this is correct and performant.